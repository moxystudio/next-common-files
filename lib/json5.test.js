'use strict';

const { withJSON5 } = require('../index.js');

const options = (overrides) => ({
    dev: true,
    isServer: true,
    ...overrides,
});

const createConfig = () => ({
    module: {
        rules: [],
    },
});

it('should configure a unique rule correctly', () => {
    const webpackConfig = withJSON5()().webpack(createConfig(), options());

    expect(webpackConfig.module.rules).toHaveLength(1);
    expect(webpackConfig.module.rules[0]).toMatchSnapshot();
});

it('should call nextConfig webpack if defined', () => {
    const nextConfig = {
        webpack: jest.fn(() => 'foo'),
    };

    const webpackConfig = withJSON5()(nextConfig).webpack(createConfig(), options());

    expect(nextConfig.webpack).toHaveBeenCalledTimes(1);
    expect(webpackConfig).toBe('foo');
});
