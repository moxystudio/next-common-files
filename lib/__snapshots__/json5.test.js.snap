// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should configure a unique rule correctly 1`] = `
Object {
  "loader": "<PROJECT_ROOT>/node_modules/json5-loader/dist/cjs.js",
  "test": /\\\\\\.json5\\$/,
}
`;
